### PR TITLE
googlechat: return the API response and use it to enrich cogito logging

### PR DESCRIPTION
**Pull request by marco-m-pix4d**
PCI-2561

### CODE REVIEW

File: cogito/gchatsink.go [hunk 1]
```diff
@@ -42,13 +42,15 @@ func (sink GoogleChatSink) Send() error {
...
+	reply, err := googlechat.TextMessage(ctx, sink.Request.Source.GChatWebHook, threadKey,
+		text)
+	if err != nil {
 		return fmt.Errorf("GoogleChatSink: %s", err)
 	}
 
 	sink.Log.Info("state posted successfully to chat",
-		"state", state, "text", text)
+		"state", state, "text", text, "space", reply.Space.DisplayName,
+		"sender", reply.Sender.DisplayName)
```
Comment (marco-m-pix4d) [inline cogito/gchatsink.go:53]: This is where the new information is used, printing human-redable chat space name and sender.

File: googlechat/googlechat.go [hunk 2]
```diff
@@ -14,33 +14,73 @@ import (
...
 )
 
-// BasicMessage represents the JSON payload of a Google Chat basic message.
+// BasicMessage is the request for a Google Chat basic message.
 type BasicMessage struct {
 	Text string `json:"text"`
 }
 
-// TextMessage sends a one-off text message with threadKey to webhook theURL.
+// MessageReply is the reply to [TextMessage].
```
Comment (marco-m-pix4d) [inline googlechat/googlechat.go:25]: This is the new information reported by TextMessage().

File: googlechat/googlechat_test.go [hunk 3]
```diff
@@ -22,30 +24,26 @@ func TestTextMessageIntegration(t *testing.T) {
...
 }
 
 func TestRedactURL(t *testing.T) {
 	hook := "https://chat.googleapis.com/v1/spaces/SSS/messages?key=KKK&token=TTT"
 	want := "https://chat.googleapis.com/v1/spaces/SSS/messages?REDACTED"
 	theURL, err := url.Parse(hook)
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, err)
```
Comment (marco-m-pix4d) [inline googlechat/googlechat_test.go:42]: also for this file I now used gotest.tools/v3/assert

File: testhelp/testserver.go [hunk 4]
```diff
@@ -9,28 +9,40 @@ import (
...
 // To avoid races, call ts.Close() before reading any parameters.
 //
 // Example:
 //
 //	var ghReq github.AddRequest
 //	var URL *url.URL
-//	ts := SpyHttpServer(&ghReq, &URL, http.StatusCreated)
-func SpyHttpServer(payload any, theUrl **url.URL, successCode int) *httptest.Server {
+//	ts := SpyHttpServer(&ghReq, nil, &URL, http.StatusCreated)
+func SpyHttpServer(request any, reply any, theUrl **url.URL, successCode int,
```
Comment (marco-m-pix4d) [inline testhelp/testserver.go:24]: The spy is now modified to also send a reply.

**Review by iAmoric — APPROVED** [review]

**Review by Esysc — APPROVED** [review]

**Review by odormond — APPROVED** [review]

**Pull request merged**